[gnome] handle GNOME 47 extensions

honnip opened the pull request:
## Description of changes

#333911

it would be good to rerun the update script later and amend it, but it's not necessary

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

honnip committed:
gnomeExtensions: handle GNOME 47 extensions

honnip committed:
gnomeExtensions: auto-update

honnip commented:
@bobby285271 I will keep this up to date

Please close when you cherry-picked

bobby285271 commented:
I never touched `pkgs/desktops/gnome/extensions` stuff before so having second pair eyes will be nice, but in any case I will ensure this get triaged before I rebase #333911.

bobby285271 committed:
gnomeExtensions.gtk4-desktop-icons-ng-ding: Fix patch

https://gitlab.com/smedius/desktop-icons-ng/-/commit/9079ac663bc33421a489dab2db231f990105bdd5

This unbreaks nixosTests.gnome-extensions.

jtojnar commented on pkgs/desktops/gnome/extensions/default.nix:
```diff
@@ -64,9 +64,10 @@ in rec {
   gnome44Extensions = mapUuidNames (produceExtensionsList "44");
   gnome45Extensions = mapUuidNames (produceExtensionsList "45");
   gnome46Extensions = mapUuidNames (produceExtensionsList "46");
+  gnome47Extensions = mapUuidNames (produceExtensionsList "47");
```
all-packages.nix should be updated too.

jtojnar reviewed (commented):
Other changes look good according to what I see in the README.

bobby285271 committed:
fixup! gnomeExtensions: handle GNOME 47 extensions

bobby285271 commented:
I plan to rebase https://github.com/NixOS/nixpkgs/pull/333911 soon so let's merge, thanks!
